Apps/W1/EDocumentConnectors/Tietoevry/app/src/Formats/FormatEvents.Codeunit.al

```diff
@@ -0,0 +1,136 @@
+    begin
+        if not IsBISBilling then
+            exit;
+        if not ConnectionSetup.Get() then
+            exit;
+
+        this.SplitId(ConnectionSetup."Company Id", SupplierSchemeID, SupplierEndpointID);
+    end;
+
+    [EventSubscriber(ObjectType::Codeunit, Codeunit::"PEPPOL Management", OnAfterGetAccountingSupplierPartyLegalEntityByFormat, '', false, false)]
+    local procedure "PEPPOL Management_OnAfterGetAccountingSupplierPartyLegalEntityByFormat"(var PartyLegalEntityRegName: Text; var PartyLegalEntityCompanyID: Text; var PartyLegalEntitySchemeID: Text; var SupplierRegAddrCityName: Text; var SupplierRegAddrCountryIdCode: Text; var SupplRegAddrCountryIdListId: Text; IsBISBilling: Boolean)
+    var
+        ConnectionSetup: Record "Connection Setup";
+    begin
+        if not IsBISBilling then
+            exit;
+        if not ConnectionSetup.Get() then
+            exit;
+
+        this.SplitId(ConnectionSetup."Company Id", PartyLegalEntitySchemeID, PartyLegalEntityCompanyID);
+    end;
+
+    [EventSubscriber(ObjectType::Codeunit, Codeunit::"PEPPOL Management", OnAfterGetAccountingCustomerPartyInfoByFormat, '', false, false)]
+    local procedure "PEPPOL Management_OnAfterGetAccountingCustomerPartyInfoByFormat"(SalesHeader: Record "Sales Header"; var CustomerEndpointID: Text; var CustomerSchemeID: Text; var CustomerPartyIdentificationID: Text; var CustomerPartyIDSchemeID: Text; var CustomerName: Text; IsBISBilling: Boolean)
+    var
+        ServiceParticipant: Record "Service Participant";
+        EDocumentService: Record "E-Document Service";
+        ConnectionSetup: Record "Connection Setup";
+    begin
+        if not IsBISBilling then
+            exit;
+        if not ConnectionSetup.Get() then
+            exit;
+        EDocumentService.SetRange("Service Integration V2", EDocumentService."Service Integration V2"::Tietoevry);
+        if not EDocumentService.FindFirst() then
+            exit;
+        ServiceParticipant.Get(EDocumentService.Code, ServiceParticipant."Participant Type"::Customer, SalesHeader."Bill-to Customer No.");
+        this.SplitId(ServiceParticipant."Participant Identifier", CustomerSchemeID, CustomerEndpointID);
+        this.SplitId(ServiceParticipant."Participant Identifier", CustomerPartyIDSchemeID, CustomerPartyIdentificationID);
+    end;
+
+    [EventSubscriber(ObjectType::Codeunit, Codeunit::"PEPPOL Management", OnAfterGetAccountingCustomerPartyLegalEntityByFormat, '', false, false)]
+    local procedure "PEPPOL Management_OnAfterGetAccountingCustomerPartyLegalEntityByFormat"(SalesHeader: Record "Sales Header"; var CustPartyLegalEntityRegName: Text; var CustPartyLegalEntityCompanyID: Text; var CustPartyLegalEntityIDSchemeID: Text; IsBISBilling: Boolean)
+    var
+        ServiceParticipant: Record "Service Participant";
+        EDocumentService: Record "E-Document Service";
+        ConnectionSetup: Record "Connection Setup";
+    begin
+        if not IsBISBilling then
```

Comment:
If I understand this event subscriber correctly then this will always change the content as soon as a connectionsetup exists and an e-document service exists for tietoevry.

Is there any chance to find out if this document will be transferred via this e document service?

If this isn't possible then there should be other mechanism to prevent conflicts with other e document services that might be installed in parallel.